include/deal.II/multigrid/mg_transfer_global_coarsening.h

```diff
@@ -529,8 +671,9 @@ class MGTransferGlobalCoarsening : public dealii::MGTransferBase<VectorType>
    * internal level vectors within the function call copy_to_mg() if used in the
    * context of PreconditionMG.
    */
+  template <typename T>
   MGTransferGlobalCoarsening(
```

Comment:
Can you please document the expectations on the template type `T`? I guess it should be possible to upcast it to `MGTwoLevelTransferBase` (or a dereferenced version if we have a `shared_ptr/unique_ptr`)?